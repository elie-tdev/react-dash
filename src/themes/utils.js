export const opacity = (hex, opacity) => {
  var result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
  if (!result) {
    throw new Error('Invalid hex', hex);
  }
  return `rgba(${parseInt(result[1], 16)}, ${parseInt(result[2], 16)}, ${parseInt(
    result[3],
    16,
  )}, ${opacity})`;
};

export const gradient = ({ type = 'linear', colors, background }) =>
  `${type}-gradient(${colors.join(', ')}), ${background};`;
