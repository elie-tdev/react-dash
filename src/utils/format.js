const formatter = new Intl.NumberFormat('en-US', {
  style: 'currency',
  currency: 'USD',
  minimumFractionDigits: 2,
});

const parseFloat = (str, val = 2) => {
  str = str && str.toString();
  str = str && str.indexOf('.') > -1 ? str.slice(0, str.indexOf('.') + val + 1) : str;
  return Number(str);
};

export const formatDollar = number => formatter.format(parseFloat(number, 2));
