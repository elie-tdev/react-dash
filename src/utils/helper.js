const maskAccountNumber = accNumber => {
  return accNumber && <>&bull;&bull;&bull;&bull; {accNumber.slice(accNumber.length - 4)}</>;
};

export default maskAccountNumber;
