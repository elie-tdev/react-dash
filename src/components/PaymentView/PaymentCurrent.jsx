import cn from 'classnames';
import { Box, Card, CardContent, makeStyles, Typography } from '@material-ui/core';
import ThumbUpIcon from '@material-ui/icons/ThumbUp';
import { useTranslation } from 'react-i18next';
import React, { useMemo, useState } from 'react';
import dateformat from 'dateformat';

import { SecondaryButton, SubmitPaymentModal } from '@/components';
import { gradient, useGlobalStyles } from '@/themes';
import { formatDateString } from '@/properties';

const useStyles = makeStyles(theme => ({
  paymentCurrentContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    display: 'flex',
    flexDirection: 'column',
  },
  icon: {
    background: gradient({
      colors: theme.palette.common.gradient.highlight,
      background: theme.palette.success.main,
    }),
    width: '48px',
    height: '48px',
    color: theme.palette.success.main,
    justifyContent: 'center',
    alignItems: 'center',
    display: 'flex',
    borderRadius: '100%',
    marginBottom: '16px',
  },
  date: {
    color: theme.palette.text.primary,
  },
  text: {
    fontSize: '14px',
    marginBottom: '32px',
    marginLeft: '32px',
    marginRight: '32px',
    display: 'flex',
    textAlign: 'center',
  },
  containedCard: {
    borderRadius: '12px',
    border: '1px solid rgba(50, 70, 118, 0.12)',
    boxSizing: 'border-box',
    boxShadow: 'none',
  },
  button: {
    height: 'auto',
  },
}));

const Container = ({ className, contained, children }) => {
  const styles = useStyles();
  if (!contained) {
    return <>{children}</>;
  }
  return (
    <Card data-testid="PaymentCurrent" className={styles.containedCard}>
      <CardContent className={className}>{children}</CardContent>
    </Card>
  );
};

const PaymentCurrent = ({ dueDate, contained, onSchedulePayment }) => {
  const styles = useStyles();
  const globalStyles = useGlobalStyles();
  const { t } = useTranslation();
  const [openSchedulePayment, setOpenSchedulePayment] = useState(false);
  const formattedDueDate = useMemo(() => dateformat(dueDate, formatDateString), [dueDate]);
  return (
    <Container className={cn(styles.paymentCurrentContainer)} contained={contained}>
      <Box className={cn(styles.icon)}>
        <ThumbUpIcon />
      </Box>
      <Typography variant="h3">{t('content.Caught up')}</Typography>
      <Typography className={cn(globalStyles.caption, styles.text)}>
        {t('content.Nice work!')}
      </Typography>
      <Box className={cn(styles.text)}>
        <Typography className={cn(globalStyles.caption)}>
          {t('content.Next Billing Date')}:&nbsp;
        </Typography>
        <Typography className={cn(globalStyles.caption, styles.date)}>
          {formattedDueDate}
        </Typography>
      </Box>
      <SecondaryButton
        fullWidth
        onClick={() => setOpenSchedulePayment(true)}
        className={styles.button}
      >
        {t('content.Schedule Additional Payment')}
      </SecondaryButton>
      <SubmitPaymentModal
        open={openSchedulePayment}
        onClose={() => setOpenSchedulePayment(false)}
        onSuccess={() => {
          setOpenSchedulePayment(false);
          onSchedulePayment();
        }}
      />
    </Container>
  );
};

export default PaymentCurrent;
