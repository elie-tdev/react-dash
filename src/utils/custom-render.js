import React from 'react';
import { I18nextProvider } from 'react-i18next';
import { render } from '@testing-library/react';
import { CssBaseline } from '@material-ui/core';
import { BrowserRouter as Router } from 'react-router-dom';
import { ThemeProvider } from '@material-ui/core/styles';
import { MockedProvider } from '@apollo/client/testing';

import { MenuContextProvider } from '@/contexts';
import i18n from '@/translations';
import { LightTheme } from '@/themes';

const wrapper = ({ mocks = [], menuContextProviderProps = {} }) => {
  const Wrapper = ({ children }) => {
    return (
      <Router>
        <MockedProvider mocks={mocks} addTypename={false}>
          <ThemeProvider theme={LightTheme}>
            <MenuContextProvider {...menuContextProviderProps}>
              <CssBaseline />
              <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
            </MenuContextProvider>
          </ThemeProvider>
        </MockedProvider>
      </Router>
    );
  };

  return Wrapper;
};

const customRender = (ui, { mocks, menuContextProviderProps, ...restOptions } = {}) =>
  render(ui, { wrapper: wrapper({ mocks, menuContextProviderProps }), ...restOptions });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
