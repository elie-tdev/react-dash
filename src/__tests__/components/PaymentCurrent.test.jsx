import React from 'react';

import { fireEvent, render } from '@/utils/custom-render';
import PaymentCurrent from '@/components/PaymentView/PaymentCurrent';

describe('Payment Current Component', () => {
  test('Renders', () => {
    const { container } = render(<PaymentCurrent contained dueDate="Jan 12, 2021" />);
    expect(container).toMatchSnapshot();
  });

  test('click on Schedule Additional Payment button', async () => {
    const handleClick = jest.fn();
    const { container, findAllByTestId } = render(
      <PaymentCurrent onClick={handleClick()} contained dueDate="Jan 12, 2021" />,
    );

    const additionalButton = await findAllByTestId('SecondaryButton');
    await fireEvent.click(additionalButton[0]);
    await expect(handleClick).toHaveBeenCalled();
    expect(container).toMatchSnapshot();
  });
});
