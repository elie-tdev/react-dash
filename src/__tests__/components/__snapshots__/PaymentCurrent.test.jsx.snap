// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment Current Component Renders 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-containedCard-5 MuiPaper-elevation1 MuiPaper-rounded"
    data-testid="PaymentCurrent"
  >
    <div
      class="MuiCardContent-root makeStyles-paymentCurrentContainer-1"
    >
      <div
        class="MuiBox-root MuiBox-root-36 makeStyles-icon-2"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M1 21h4V9H1v12zm22-11c0-1.1-.9-2-2-2h-6.31l.95-4.57.03-.32c0-.41-.17-.79-.44-1.06L14.17 1 7.59 7.59C7.22 7.95 7 8.45 7 9v10c0 1.1.9 2 2 2h9c.83 0 1.54-.5 1.84-1.22l3.02-7.05c.09-.23.14-.47.14-.73v-2z"
          />
        </svg>
      </div>
      <h3
        class="MuiTypography-root MuiTypography-h3"
      >
        You're all caught up!
      </h3>
      <p
        class="MuiTypography-root makeStyles-caption-13 makeStyles-text-4 MuiTypography-body1"
      >
        Nice work! You are current on all your loan payments. This does not reflect scheduled or pending payments.
      </p>
      <div
        class="MuiBox-root MuiBox-root-37 makeStyles-text-4"
      >
        <p
          class="MuiTypography-root makeStyles-caption-13 MuiTypography-body1"
        >
          Next Billing Date
          : 
        </p>
        <p
          class="MuiTypography-root makeStyles-caption-13 makeStyles-date-3 MuiTypography-body1"
        >
          Jan 12, 2021
        </p>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root makeStyles-root-38 makeStyles-root-39 MuiButton-contained makeStyles-button-6 MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation MuiButton-fullWidth"
        data-testid="SecondaryButton"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Schedule Additional Payment
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        aria-hidden="true"
        class="MuiBackdrop-root makeStyles-confirmPanelBackdrop-68"
        style="opacity: 0; visibility: hidden;"
      />
      <div
        aria-hidden="true"
        class="MuiBackdrop-root makeStyles-confirmPanelBackdrop-72"
        style="opacity: 0; visibility: hidden;"
      />
      <div
        aria-hidden="true"
        class="MuiBackdrop-root makeStyles-confirmPanelBackdrop-84"
        style="opacity: 0; visibility: hidden;"
      />
      <div
        aria-hidden="true"
        class="MuiDrawer-root MuiDrawer-modal makeStyles-confirmPanelRoot-95"
        role="presentation"
        style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px; visibility: hidden;"
      >
        <div
          aria-hidden="true"
          class="MuiBackdrop-root"
          style="opacity: 0; visibility: hidden;"
        />
        <div
          data-test="sentinelStart"
          tabindex="0"
        />
        <div
          class="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorBottom makeStyles-modalPaper-35 MuiPaper-elevation16"
          style="visibility: hidden; webkit-transform: translateY(768px) translateY(0px); transform: translateY(768px) translateY(0px);"
          tabindex="-1"
        >
          <div
            class="MuiBox-root MuiBox-root-104 makeStyles-header-100 makeStyles-row-14"
          >
            <div
              class="MuiBox-root MuiBox-root-105 makeStyles-arrowBackBox-102"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                />
              </svg>
            </div>
            <p
              class="MuiTypography-root makeStyles-headerTitle-101 MuiTypography-body1"
            >
              New Payment Account
            </p>
            <div
              class="MuiBox-root MuiBox-root-106"
            />
          </div>
          <hr
            class="MuiDivider-root"
          />
          <div
            class="MuiDialogContent-root"
          >
            <div
              class="MuiFormControl-root makeStyles-inputField-91 MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputLabel-92 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused"
                data-shrink="true"
              >
                Routing Number
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root makeStyles-newPaymentMethodInput-93 MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
                variant="outlined"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="routingNum"
                  placeholder="Routing Number"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-107 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-109"
                  >
                    <span>
                      Routing Number
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root makeStyles-inputField-91 MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputLabel-92 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
                data-shrink="true"
              >
                Account Number
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root makeStyles-newPaymentMethodInput-93 MuiInputBase-fullWidth MuiInputBase-formControl"
                variant="outlined"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="accountNum"
                  placeholder="Account Number"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-107 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-109"
                  >
                    <span>
                      Account Number
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root makeStyles-inputField-91 MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputLabel-92 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
                data-shrink="true"
              >
                Verify Account Number
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root makeStyles-newPaymentMethodInput-93 MuiInputBase-fullWidth MuiInputBase-formControl"
                variant="outlined"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="accountNumVerify"
                  placeholder="Verify Account Number"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-107 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-109"
                  >
                    <span>
                      Verify Account Number
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root makeStyles-inputFieldLast-96 MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputFieldLastLabel-97 makeStyles-inputLabel-92 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
                data-shrink="true"
              >
                Account Type
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <div
                  aria-haspopup="listbox"
                  class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                  role="button"
                  tabindex="0"
                >
                  <span>
                    ​
                  </span>
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput"
                  tabindex="-1"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-107 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-108"
                    style="width: 0.01px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiPaper-root MuiCard-root makeStyles-saveAsDefaultCard-98 MuiPaper-elevation0 MuiPaper-rounded"
            >
              <div
                class="MuiBox-root MuiBox-root-111 makeStyles-saveAsDefaultBox-99"
              >
                <p
                  class="MuiTypography-root makeStyles-inputLabel-92 MuiTypography-body1"
                >
                  Save as default payment account
                </p>
                <span
                  class="MuiSwitch-root"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-112 MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-checked-113 Mui-checked"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <input
                        checked=""
                        class="PrivateSwitchBase-input-115 MuiSwitch-input"
                        type="checkbox"
                        value=""
                      />
                      <span
                        class="MuiSwitch-thumb"
                      />
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiSwitch-track"
                  />
                </span>
              </div>
            </div>
            <div
              class="MuiBox-root MuiBox-root-116"
            >
              <button
                class="MuiButtonBase-root MuiButton-root makeStyles-root-117 makeStyles-root-118 MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation Mui-disabled MuiButton-fullWidth Mui-disabled"
                data-testid="PrimaryButton"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Add New Account
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          data-test="sentinelEnd"
          tabindex="0"
        />
      </div>
      <div
        aria-hidden="true"
        class="MuiBackdrop-root makeStyles-confirmPanelBackdrop-122"
        style="opacity: 0; visibility: hidden;"
      />
    </div>
  </div>
</div>
`;

exports[`Payment Current Component click on Schedule Additional Payment button 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-containedCard-151 MuiPaper-elevation1 MuiPaper-rounded"
    data-testid="PaymentCurrent"
  >
    <div
      class="MuiCardContent-root makeStyles-paymentCurrentContainer-147"
    >
      <div
        class="MuiBox-root MuiBox-root-182 makeStyles-icon-148"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M1 21h4V9H1v12zm22-11c0-1.1-.9-2-2-2h-6.31l.95-4.57.03-.32c0-.41-.17-.79-.44-1.06L14.17 1 7.59 7.59C7.22 7.95 7 8.45 7 9v10c0 1.1.9 2 2 2h9c.83 0 1.54-.5 1.84-1.22l3.02-7.05c.09-.23.14-.47.14-.73v-2z"
          />
        </svg>
      </div>
      <h3
        class="MuiTypography-root MuiTypography-h3"
      >
        You're all caught up!
      </h3>
      <p
        class="MuiTypography-root makeStyles-caption-159 makeStyles-text-150 MuiTypography-body1"
      >
        Nice work! You are current on all your loan payments. This does not reflect scheduled or pending payments.
      </p>
      <div
        class="MuiBox-root MuiBox-root-183 makeStyles-text-150"
      >
        <p
          class="MuiTypography-root makeStyles-caption-159 MuiTypography-body1"
        >
          Next Billing Date
          : 
        </p>
        <p
          class="MuiTypography-root makeStyles-caption-159 makeStyles-date-149 MuiTypography-body1"
        >
          Jan 12, 2021
        </p>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root makeStyles-root-184 makeStyles-root-185 MuiButton-contained makeStyles-button-152 MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation MuiButton-fullWidth"
        data-testid="SecondaryButton"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Schedule Additional Payment
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        aria-hidden="true"
        class="MuiBackdrop-root makeStyles-backdrop-180"
        style="opacity: 1; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; webkit-transform: none; transform: none;"
      >
        <div
          class="MuiDrawer-root makeStyles-root-199 MuiDrawer-modal"
          id="submitPaymenModalt"
          role="presentation"
          style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
        >
          <div
            aria-hidden="true"
            class="MuiBackdrop-root"
            style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
          />
          <div
            data-test="sentinelStart"
            tabindex="0"
          />
          <div
            class="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorBottom makeStyles-modalPaper-181 MuiPaper-elevation16"
            data-testid="SubmitPayment"
            tabindex="-1"
          >
            <div
              class="MuiBox-root MuiBox-root-293 makeStyles-header-200 makeStyles-row-160"
            >
              <div
                class="MuiBox-root MuiBox-root-294"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </div>
              <p
                class="MuiTypography-root makeStyles-headerTitle-204 MuiTypography-body1"
              >
                Schedule a Payment
              </p>
              <div
                class="MuiBox-root MuiBox-root-295"
              />
            </div>
            <hr
              class="MuiDivider-root"
            />
            <div
              class="MuiBox-root MuiBox-root-296"
            >
              <div
                class="MuiDialogContent-root makeStyles-dialogContent-207"
              >
                <h6
                  class="MuiTypography-root makeStyles-selectAmountTitle-189 MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
                >
                  Select your payment amount
                </h6>
                <div
                  class="MuiBox-root MuiBox-root-308 makeStyles-fauxInputField-301"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
                  >
                    Regular Payment Amount
                  </p>
                  <p
                    class="MuiTypography-root makeStyles-fauxFloatRight-304 MuiTypography-body1"
                  />
                  <div
                    class="MuiPaper-root MuiAlert-root MuiAlert-standardSuccess makeStyles-alert-306 MuiPaper-elevation0"
                    role="alert"
                  >
                    <div
                      class="MuiAlert-message"
                    >
                      Recommended
                    </div>
                  </div>
                  <br />
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
                  />
                </div>
                <div
                  class="MuiBox-root MuiBox-root-309 makeStyles-fauxInputField-301"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 MuiTypography-colorPrimary"
                  >
                    Other Amount
                  </p>
                  <p
                    class="MuiTypography-root makeStyles-fauxFloatRight-304 MuiTypography-body1"
                  />
                  <br />
                  <div
                    class="MuiInputBase-root makeStyles-additionalAmountInput-298 MuiInputBase-fullWidth"
                    variant=""
                  >
                    <input
                      class="MuiInputBase-input"
                      id="otherAmount"
                      maxlength="6"
                      placeholder="$0"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <p
                  class="MuiTypography-root makeStyles-lowerText-186 MuiTypography-body1 MuiTypography-colorTextSecondary"
                />
              </div>
            </div>
          </div>
          <div
            data-test="sentinelEnd"
            tabindex="0"
          />
        </div>
      </div>
      <div
        aria-hidden="true"
        class="MuiBackdrop-root makeStyles-confirmPanelBackdrop-325"
        style="opacity: 0; visibility: hidden;"
      />
      <div
        aria-hidden="true"
        class="MuiBackdrop-root makeStyles-confirmPanelBackdrop-336"
        style="opacity: 0; visibility: hidden;"
      />
      <div
        aria-hidden="true"
        class="MuiBackdrop-root makeStyles-confirmPanelBackdrop-214"
        style="opacity: 0; visibility: hidden;"
      />
      <div
        aria-hidden="true"
        class="MuiBackdrop-root makeStyles-confirmPanelBackdrop-218"
        style="opacity: 0; visibility: hidden;"
      />
      <div
        aria-hidden="true"
        class="MuiBackdrop-root makeStyles-confirmPanelBackdrop-230"
        style="opacity: 0; visibility: hidden;"
      />
      <div
        aria-hidden="true"
        class="MuiDrawer-root MuiDrawer-modal makeStyles-confirmPanelRoot-241"
        role="presentation"
        style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px; visibility: hidden;"
      >
        <div
          aria-hidden="true"
          class="MuiBackdrop-root"
          style="opacity: 0; visibility: hidden;"
        />
        <div
          data-test="sentinelStart"
          tabindex="0"
        />
        <div
          class="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorBottom makeStyles-modalPaper-181 MuiPaper-elevation16"
          style="visibility: hidden; webkit-transform: translateY(768px) translateY(0px); transform: translateY(768px) translateY(0px);"
          tabindex="-1"
        >
          <div
            class="MuiBox-root MuiBox-root-250 makeStyles-header-246 makeStyles-row-160"
          >
            <div
              class="MuiBox-root MuiBox-root-251 makeStyles-arrowBackBox-248"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                />
              </svg>
            </div>
            <p
              class="MuiTypography-root makeStyles-headerTitle-247 MuiTypography-body1"
            >
              New Payment Account
            </p>
            <div
              class="MuiBox-root MuiBox-root-252"
            />
          </div>
          <hr
            class="MuiDivider-root"
          />
          <div
            class="MuiDialogContent-root"
          >
            <div
              class="MuiFormControl-root makeStyles-inputField-237 MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputLabel-238 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
                data-shrink="true"
              >
                Routing Number
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root makeStyles-newPaymentMethodInput-239 MuiInputBase-fullWidth MuiInputBase-formControl"
                variant="outlined"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="routingNum"
                  placeholder="Routing Number"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-253 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-255"
                  >
                    <span>
                      Routing Number
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root makeStyles-inputField-237 MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputLabel-238 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
                data-shrink="true"
              >
                Account Number
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root makeStyles-newPaymentMethodInput-239 MuiInputBase-fullWidth MuiInputBase-formControl"
                variant="outlined"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="accountNum"
                  placeholder="Account Number"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-253 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-255"
                  >
                    <span>
                      Account Number
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root makeStyles-inputField-237 MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputLabel-238 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
                data-shrink="true"
              >
                Verify Account Number
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root makeStyles-newPaymentMethodInput-239 MuiInputBase-fullWidth MuiInputBase-formControl"
                variant="outlined"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  id="accountNumVerify"
                  placeholder="Verify Account Number"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-253 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-255"
                  >
                    <span>
                      Verify Account Number
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root makeStyles-inputFieldLast-242 MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root makeStyles-inputFieldLastLabel-243 makeStyles-inputLabel-238 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
                data-shrink="true"
              >
                Account Type
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <div
                  aria-haspopup="listbox"
                  class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                  role="button"
                  tabindex="0"
                >
                  <span>
                    ​
                  </span>
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput"
                  tabindex="-1"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-253 MuiOutlinedInput-notchedOutline"
                  style="padding-left: 8px;"
                >
                  <legend
                    class="PrivateNotchedOutline-legend-254"
                    style="width: 0.01px;"
                  >
                    <span>
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiPaper-root MuiCard-root makeStyles-saveAsDefaultCard-244 MuiPaper-elevation0 MuiPaper-rounded"
            >
              <div
                class="MuiBox-root MuiBox-root-257 makeStyles-saveAsDefaultBox-245"
              >
                <p
                  class="MuiTypography-root makeStyles-inputLabel-238 MuiTypography-body1"
                >
                  Save as default payment account
                </p>
                <span
                  class="MuiSwitch-root"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-258 MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-checked-259 Mui-checked"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <input
                        checked=""
                        class="PrivateSwitchBase-input-261 MuiSwitch-input"
                        type="checkbox"
                        value=""
                      />
                      <span
                        class="MuiSwitch-thumb"
                      />
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    class="MuiSwitch-track"
                  />
                </span>
              </div>
            </div>
            <div
              class="MuiBox-root MuiBox-root-262"
            >
              <button
                class="MuiButtonBase-root MuiButton-root makeStyles-root-263 makeStyles-root-264 MuiButton-contained MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-disableElevation Mui-disabled MuiButton-fullWidth Mui-disabled"
                data-testid="PrimaryButton"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Add New Account
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          data-test="sentinelEnd"
          tabindex="0"
        />
      </div>
      <div
        aria-hidden="true"
        class="MuiBackdrop-root makeStyles-confirmPanelBackdrop-268"
        style="opacity: 0; visibility: hidden;"
      />
    </div>
  </div>
</div>
`;
